### DISCUSSION TITLE
Training YoloV8 using amd rocm

### DISCUSSION THREAD

**Discussion by rodrigoandrigo** [ultralytics/ultralytics#15781] [General]
rocm installation tutorial:
https://www.youtube.com/watch?v=hBMvM9eQhPs

use this at the beginning of the code to avoid the error: "RuntimeError: HIP error: invalid device function"

from os import putenv
putenv("HSA_OVERRIDE_GFX_VERSION", "10.3.0") # The line must be defined before importing torch.

import torch # OK. The HIP Runtime of PyTorch can recognize your ISA.

![Captura de tela de 2024-08-23 09-40-33](https://github.com/user-attachments/assets/7bf9f830-9950-4e1f-81aa-adc2e63a9e67)

**Comment by glenn-jocher** [comment]
@rodrigoandrigo thank you for sharing this information. If you encounter any issues while training YOLOv8 using AMD ROCm, please ensure you are using the latest version of the Ultralytics package. If the issue persists, feel free to provide more details, and we'll be happy to assist further.
